actionpack/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb

```diff
@@ -42,7 +42,10 @@ def take_screenshot(html: false, screenshot: nil)
         #
         # +take_failed_screenshot+ is called during system test teardown.
         def take_failed_screenshot
-          take_screenshot if failed? && supports_screenshot? && Capybara::Session.instance_created?
+          return unless failed? && supports_screenshot? && Capybara::Session.instance_created?
+
+          take_screenshot
+          metadata[:failure_screenshot_path] = absolute_image_path if defined?(metadata)
```

Comment:
```suggestion
          metadata[:failure_screenshot_path] = absolute_image_path if Minitest::Runnable.method_defined?(:metadata)
```

Otherwise this code could be confused by a user defined `metadata` method in the test class.

Reply:
It makes sense being more explicit when checking for this. I applied your suggestion. Thanks.

Reply:
This now fails when minitest isn't the test runner, for example when rspec is the runner. Is there an additional condition that can be added to detect that minitest is the runner and not just loaded?

Reply:
@albus522 Does the test fail or is there just no additional metadata? Could you open an issue with a repro? :bow:

Reply:
@zzak @matteeyah I reproduced the error reported by @albus522 with a video.
Let me know how I can help solving this. Thanks 🙇 

https://github.com/rails/rails/assets/10076/eb60d0fe-0ca2-4986-a5d2-bce9faf2928d

Reply:
This was fixed today by @JonRowe in https://github.com/rspec/rspec-rails/pull/2704 🎉

Reply:
Thanks for the thorough explanation! Glad it was fixed downstream. 🙏